src/hotspot/share/cds/heapShared.cpp

```diff
@@ -587,7 +589,13 @@ void HeapShared::copy_closed_objects() {
 void HeapShared::copy_open_objects() {
   assert(HeapShared::can_write(), "must be");
 
-  archive_java_mirrors();
+  {
```

Comment:
Why do  you need a new  scope here?

Reply:
I used the scope to mark that the two archiving operations must be done within the scope of

```
{
    init_seen_objects_table();
       ....
    delete_seen_objects_table();
}
```

And the `delete_seen_objects_table()` must be called before the rest of the function can proceed.